app/src/main/java/jimpatrizi/com/netrtl/MainActivity.java

```diff
@@ -602,10 +629,21 @@ public void threadClientInit() {
          * Obtain the shared preferences object and get the respective keys for user set ip addr and port
          */
         sharedPrefs = context.getSharedPreferences("pref_main", Context.MODE_PRIVATE);
-        //inits to 0.0.0 if field is empty
-        ip_address = sharedPrefs.getString("key_ip_name", "0.0.0");
+
+        // Returns null if no value has been set to this shared preferences key
+        ipAddress = sharedPrefs.getString("key_ip_name", null);
+
         //inits port to 2832 if field is empty
-        port_number = sharedPrefs.getInt("key_port_name", 2832);
+        portNumber = sharedPrefs.getInt("key_port_name", 2832);
+
+        // If no value has been set for the IP, don't try to connect.
+        if (ipAddress == null || ipAddress.length() == 0)
+        {
+            showGotItDialog("Connection information", "Visit the Settings page to set a " +
```

Comment:
Ideally, the popup should bring you to the settings page

Reply:
ill try to remember this, but I should be able to do this